File: openapi-generator/java-play-framework/app/apimodels/Pagination.java
package apimodels;

import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * Pagination
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2019-03-01T05:34:42.270Z[GMT]")

@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class Pagination   {
  @JsonProperty("endCursor")
  private String endCursor;

  @JsonProperty("hasNextPage")
  private Boolean hasNextPage;

  @JsonProperty("hasPrevPage")
  private Boolean hasPrevPage;

  @JsonProperty("startCursor")
  private String startCursor;

  public Pagination endCursor(String endCursor) {
    this.endCursor = endCursor;
    return this;
  }

   /**
   * Cursor identifier representing the last element in the response. This value should be used in conjunction with a subsequent request's 'starting_after' query parameter.
   * @return endCursor
  **/
  @NotNull
  public String getEndCursor() {
    return endCursor;
  }

  public void setEndCursor(String endCursor) {
    this.endCursor = endCursor;
  }

  public Pagination hasNextPage(Boolean hasNextPage) {
    this.hasNextPage = hasNextPage;
    return this;
  }

   /**
   * True if there are more pages of results after this response.
   * @return hasNextPage
  **/
  @NotNull
  public Boolean getHasNextPage() {
    return hasNextPage;
  }

  public void setHasNextPage(Boolean hasNextPage) {
    this.hasNextPage = hasNextPage;
  }

  public Pagination hasPrevPage(Boolean hasPrevPage) {
    this.hasPrevPage = hasPrevPage;
    return this;
  }

   /**
   * True if there are more pages of results before this response.
   * @return hasPrevPage
  **/
  @NotNull
  public Boolean getHasPrevPage() {
    return hasPrevPage;
  }

  public void setHasPrevPage(Boolean hasPrevPage) {
    this.hasPrevPage = hasPrevPage;
  }

  public Pagination startCursor(String startCursor) {
    this.startCursor = startCursor;
    return this;
  }

   /**
   * Cursor identifier representing the first element in the response. This value should be used in conjunction with a subsequent request's 'ending_before' query parameter.
   * @return startCursor
  **/
  @NotNull
  public String getStartCursor() {
    return startCursor;
  }

  public void setStartCursor(String startCursor) {
    this.startCursor = startCursor;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Pagination pagination = (Pagination) o;
    return Objects.equals(endCursor, pagination.endCursor) &&
        Objects.equals(hasNextPage, pagination.hasNextPage) &&
        Objects.equals(hasPrevPage, pagination.hasPrevPage) &&
        Objects.equals(startCursor, pagination.startCursor);
  }

  @Override
  public int hashCode() {
    return Objects.hash(endCursor, hasNextPage, hasPrevPage, startCursor);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Pagination {\n");
    
    sb.append("    endCursor: ").append(toIndentedString(endCursor)).append("\n");
    sb.append("    hasNextPage: ").append(toIndentedString(hasNextPage)).append("\n");
    sb.append("    hasPrevPage: ").append(toIndentedString(hasPrevPage)).append("\n");
    sb.append("    startCursor: ").append(toIndentedString(startCursor)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: openapi-generator/csharp-nancyfx/src/Org.OpenAPITools/Models/Pagination.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools.v1.Models
{
    /// <summary>
    /// Pagination
    /// </summary>
    public sealed class Pagination:  IEquatable<Pagination>
    { 
        /// <summary>
        /// Cursor identifier representing the last element in the response. This value should be used in conjunction with a subsequent request&#39;s &#39;starting_after&#39; query parameter.
        /// </summary>
        public string EndCursor { get; private set; }

        /// <summary>
        /// True if there are more pages of results after this response.
        /// </summary>
        public bool? HasNextPage { get; private set; }

        /// <summary>
        /// True if there are more pages of results before this response.
        /// </summary>
        public bool? HasPrevPage { get; private set; }

        /// <summary>
        /// Cursor identifier representing the first element in the response. This value should be used in conjunction with a subsequent request&#39;s &#39;ending_before&#39; query parameter.
        /// </summary>
        public string StartCursor { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use Pagination.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public Pagination()
        {
        }

        private Pagination(string EndCursor, bool? HasNextPage, bool? HasPrevPage, string StartCursor)
        {
            
            this.EndCursor = EndCursor;
            
            this.HasNextPage = HasNextPage;
            
            this.HasPrevPage = HasPrevPage;
            
            this.StartCursor = StartCursor;
            
        }

        /// <summary>
        /// Returns builder of Pagination.
        /// </summary>
        /// <returns>PaginationBuilder</returns>
        public static PaginationBuilder Builder()
        {
            return new PaginationBuilder();
        }

        /// <summary>
        /// Returns PaginationBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>PaginationBuilder</returns>
        public PaginationBuilder With()
        {
            return Builder()
                .EndCursor(EndCursor)
                .HasNextPage(HasNextPage)
                .HasPrevPage(HasPrevPage)
                .StartCursor(StartCursor);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(Pagination other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (Pagination.
        /// </summary>
        /// <param name="left">Compared (Pagination</param>
        /// <param name="right">Compared (Pagination</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (Pagination left, Pagination right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (Pagination.
        /// </summary>
        /// <param name="left">Compared (Pagination</param>
        /// <param name="right">Compared (Pagination</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (Pagination left, Pagination right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of Pagination.
        /// </summary>
        public sealed class PaginationBuilder
        {
            private string _EndCursor;
            private bool? _HasNextPage;
            private bool? _HasPrevPage;
            private string _StartCursor;

            internal PaginationBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for Pagination.EndCursor property.
            /// </summary>
            /// <param name="value">Cursor identifier representing the last element in the response. This value should be used in conjunction with a subsequent request&#39;s &#39;starting_after&#39; query parameter.</param>
            public PaginationBuilder EndCursor(string value)
            {
                _EndCursor = value;
                return this;
            }

            /// <summary>
            /// Sets value for Pagination.HasNextPage property.
            /// </summary>
            /// <param name="value">True if there are more pages of results after this response.</param>
            public PaginationBuilder HasNextPage(bool? value)
            {
                _HasNextPage = value;
                return this;
            }

            /// <summary>
            /// Sets value for Pagination.HasPrevPage property.
            /// </summary>
            /// <param name="value">True if there are more pages of results before this response.</param>
            public PaginationBuilder HasPrevPage(bool? value)
            {
                _HasPrevPage = value;
                return this;
            }

            /// <summary>
            /// Sets value for Pagination.StartCursor property.
            /// </summary>
            /// <param name="value">Cursor identifier representing the first element in the response. This value should be used in conjunction with a subsequent request&#39;s &#39;ending_before&#39; query parameter.</param>
            public PaginationBuilder StartCursor(string value)
            {
                _StartCursor = value;
                return this;
            }


            /// <summary>
            /// Builds instance of Pagination.
            /// </summary>
            /// <returns>Pagination</returns>
            public Pagination Build()
            {
                Validate();
                return new Pagination(
                    EndCursor: _EndCursor,
                    HasNextPage: _HasNextPage,
                    HasPrevPage: _HasPrevPage,
                    StartCursor: _StartCursor
                );
            }

            private void Validate()
            { 
                if (_EndCursor == null)
                {
                    throw new ArgumentException("EndCursor is a required property for Pagination and cannot be null");
                } 
                if (_HasNextPage == null)
                {
                    throw new ArgumentException("HasNextPage is a required property for Pagination and cannot be null");
                } 
                if (_HasPrevPage == null)
                {
                    throw new ArgumentException("HasPrevPage is a required property for Pagination and cannot be null");
                } 
                if (_StartCursor == null)
                {
                    throw new ArgumentException("StartCursor is a required property for Pagination and cannot be null");
                } 
            }
        }

        
    }
}